napari/layers/_shapes_layer/shape_util.py

```diff
@@ -3,6 +3,32 @@
 from vispy.geometry import PolygonData
 
 
+def slice_by_plane(data):
+    """Determine if shape is entirely definied within a single 2D plane
+
+    Parameters
+    ----------
+    data : np.ndarray
+        NxD array of vertices specifying the shape.
+
+    Returns
+    ----------
+    key : str | tuple of int | bool
+        If shape is 2D return the string '2D', otherwise return the tuple
+        of int defining the 2D plane if the shape is entirely definied
+        within a single 2D plane. Otherwise return False
+    value : np.ndarray | False
+        Nx2 array of vertices specifying the shape in the 2D plane or False
+        if no such specification exists.
+    """
+    if data.shape[1] == 2:
+        return '2D', data
+    elif (data[:, :-2] == data[0, :-2]).all():
+        return tuple(data[0, :-2]), data[:, -2:]
+    else:
+        return False, False
```

Comment:
I don't know where we are at with the dims model. Does it already encapsulate the concept of which dimensions are drawn? If it does, then this function should directly look at that information, and pick out the right dimensions to look at (rather than assume it's the leading dimension). If it doesn't, then a `# TODO` here would be most useful!

Reply:
The key could then be a tuple: `(slice_dims, coords)`.

Reply:
agreed, that is not currently supported in the dims model, but I have added a `#TODO` with all this information and we can revisit once it is supported